packages/backend/src/index.ts

```diff
@@ -35,20 +37,26 @@ import scaffolder from './plugins/scaffolder';
 import sentry from './plugins/sentry';
 import { PluginEnvironment } from './types';
 
-function createEnv(plugin: string): PluginEnvironment {
-  const logger = getRootLogger().child({ type: 'plugin', plugin });
-  const database = knex({
-    client: 'sqlite3',
-    connection: ':memory:',
-    useNullAsDefault: true,
-  });
-  database.client.pool.on('createSuccess', (_eventId: any, resource: any) => {
-    resource.run('PRAGMA foreign_keys = ON', () => {});
-  });
-  return { logger, database };
+function makeCreateEnv(loadedConfigs: AppConfig[]) {
+  const config = ConfigReader.fromConfigs(loadedConfigs);
```

Comment:
This is about to become different configs based on the plugin name, which is why this code is jumping through some hoops it wouldn't need to otherwise.